### PR TITLE
feat: allow to exclude routes from SSR

### PR DESCRIPTION
**Pull request by schiller-manuel**

### CODE REVIEW

File: packages/react-router/src/lazyRouteComponent.tsx [hunk 1]
```diff
@@ -77,10 +101,21 @@ export function lazyRouteComponent<
       throw error
     }
 
-    if (!comp) {
-      throw load()
+    if (typeof window === 'undefined' && ssr?.() === false) {
```
Comment (tannerlinsley) [inline packages/react-router/src/lazyRouteComponent.tsx:104]: Should we use document here?

Comment (schiller-manuel) [reply packages/react-router/src/lazyRouteComponent.tsx]: done

**Review by SeanCassiere — COMMENTED** [review]
Since the usage for this looks to be setting `ssr` to `false` on the `Route.options`, should we have a `defaultSsr` flag on the `RouterOptions` as well?

So that'd allow both opting out certain routes from SSR as well as the choice of being able to opt-in for SSR on certain routes.

Or conversely, should this maybe be something named like `renderMode`? where maybe in the future `RSC` would be slotted in. - `"csr" | "ssr" | "rsc"`

**Comment by schiller-manuel** [comment]
the case against a default value is that we don't have access to the router instance here. so I would not add this right now

**Pull request merged**